unittest/server/src/ivi_input_controller_uinttests.cpp

```diff
@@ -0,0 +1,1714 @@
+ *                         +# Allocate memory for resources are freed when running the test
+ */
+TEST_F(InputControllerTest, handle_surface_destroy_success)
+{
+    handle_surface_destroy(&mp_ctxInput->surface_destroyed, &mp_iviSurface[0]);
+
+    ASSERT_EQ(wl_list_remove_fake.call_count, 1);
+
+    mpp_seatFocus[0] = (struct seat_focus*)calloc(1, sizeof(struct seat_focus));
+}
+
+/** ================================================================================================
+ * @test_id             handle_surface_create_nullSeatCtx
+ * @brief               Test case of handle_surface_create() where invalid input params
+ * @test_procedure Steps:
+ *                      -# Create input context and set seat_list for input context
+ *                      -# Calling the handle_surface_create() with input surface is null pointer
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() not be called
+ */
+TEST_F(InputControllerTest, handle_surface_create_nullSeatCtx)
+{
+    struct input_context l_input_ctx;
+    custom_wl_list_init(&l_input_ctx.seat_list);
+    l_input_ctx.ivishell = &m_iviShell;
+    l_input_ctx.seat_default_name = (char*)mp_seatName[0];
+
+    handle_surface_create(&l_input_ctx.surface_created, &mp_iviSurface[0]);
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 0);
+}
+
+/** ================================================================================================
+ * @test_id             handle_surface_create_success
+ * @brief               Test case of handle_surface_create() where valid input params
+ * @test_procedure Steps:
+ *                      -# Calling the handle_surface_create() with valid input context and input surface
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() must be called 2 times
+ */
+TEST_F(InputControllerTest, handle_surface_create_success)
+{
+    mp_ctxInput->seat_default_name = (char*)mp_seatName[0];
+    handle_surface_create(&mp_ctxInput->surface_created, &mp_iviSurface[0]);
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 2);
+}
+
+/** ================================================================================================
+ * @test_id             unbind_resource_controller_success
+ * @brief               Test case of unbind_resource_controller() where valid input params
+ * @test_procedure Steps:
+ *                      -# Calling the unbind_resource_controller()
+ *                      -# Verification point:
+ *                         +# wl_list_remove() must be called once time
+ */
+TEST_F(InputControllerTest, unbind_resource_controller_success)
+{
+    enable_utility_funcs_of_array_list();
+    wl_resource_from_link_fake.custom_fake = custom_wl_resource_from_link;
+    wl_resource_get_link_fake.custom_fake = custom_wl_resource_get_link;
+    unbind_resource_controller(&(mp_wlResource[0]));
+    ASSERT_EQ(wl_list_remove_fake.call_count, 1);
```

Comment:
after unbind, we expect the resource_list of input context don't contain this input resource. consider to add more verification point